content/en/docs/tasks/security/automtls/index.md

```diff
@@ -0,0 +1,276 @@
+---
+title: Automatic mutual TLS
+description: A simplified workflow to adopt mutual TLS with minimal configuration overhead.
+weight: 10
+keywords: [security,mtls,ux]
+---
+
+This tasks shows a simplified workflow for mutual TLS adoption[authentication overview](/docs/concepts/security/#authentication).
+
+With Istio auto mutual TLS feature, you can adopt mutual TLS by only configuring Authentication Policy without worrying about destination rule.
+
+Istio tracks the server workloads are migrated to Istio sidecar, and configure client sidecar to send mutual TLS traffic to those workloads automatically, and send plain text traffic to workloads
+without sidecars. This allows you to adopt Istio mutual TLS incrementally with minimal manual configuration.
+
+## Before you begin
+
+* Understand Istio [authentication policy](/docs/concepts/security/#authentication-policies) and related
+[mutual TLS authentication](/docs/concepts/security/#mutual-tls-authentication) concepts.
+
+* Have a Kubernetes cluster with Istio installed (e.g use `install/kubernetes/istio-demo.yaml` as described in
+[installation steps](/docs/setup/install/kubernetes/#installation-steps), set `global.mtls.enabled` to false  and `global.mtls.auto` to true using [Helm](/docs/setup/install/helm/)). For example,
+
+{{< text bash >}}
+$ helm template install/kubernetes/helm/istio --name istio --namespace istio-system  \
+  --set global.mtls.enabled=false  --set global.mtls.auto=true \
+  --values install/kubernetes/helm/istio/values-istio-demo.yaml | kubectl apply -f -
+{{< /text >}}
+
+## Instructions
+
+### Setup
+
+Our examples use three namespaces, `foo`, `mixed`, and `legacy`. 
+
+Each namespace we deploy `httpbin` and `sleep`.
+
+All workloads in `foo` namespace have sidecar.
+
+{{< text bash >}}
+$ kubectl create ns foo
+$ kubectl apply -f <(istioctl kube-inject -f @samples/httpbin/httpbin.yaml@) -n foo
+$ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@) -n foo
+{{< /text >}}
+
+Some workloads in `mixed` have sidecar and some don't. 
+
+{{< text bash >}}
+$ kubectl create ns mixed
+$ kubectl apply -f <(istioctl kube-inject -f @samples/httpbin/httpbin.yaml@) -n mixed
+$ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@) -n mixed
+$ cat <<EOF | kubectl apply -n mixed -f -
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: httpbin
```

Comment:
This is the same deployment as on line 49, isn't it? Won't this replace the previous one, so the pod with sidecar will be deleted?

Why do we need both anyway? What special is it demonstrating?

Reply:
This deployment is created inline, without sidecar injected, i made the changes on the label. This won't replace previous deployment (which has sidecar and different app labels).

But you're right, probably i need to modify the deployment name as well.

I'm trying to demonstrate customers don't have to worry about configuration with mTLS, even when their server workloads are partially sidecar injected. Compared with `foo` namespace

Reply:
modified the name to be `httpbin-nosidecar` this should be more obvious now.